typing_requirements.txt

```diff
@@ -0,0 +1,8 @@
+mypy
```

Comment:
why add a specific file for the typing requirements?

Reply:
it's used in the mypy workflow.
and we need them while working on local to verify mypy issues.